sonar-plugin-api/src/main/java/org/sonar/api/server/profile/BuiltInQualityProfilesDefinition.java

```diff
@@ -0,0 +1,419 @@
+     */
+    NewBuiltInQualityProfile setDefault(boolean value);
+
+    /**
+     * Activate a rule with specified key.
+     *
+     * @throws IllegalArgumentException if rule is already activated in this profile.
+     */
+    NewBuiltInActiveRule activateRule(String repoKey, String ruleKey);
+
+    Collection<NewBuiltInActiveRule> activeRules();
+
+    String language();
+
+    String name();
+
+    boolean isDefault();
+
+    void done();
+  }
+
+  class NewBuiltInQualityProfileImpl implements NewBuiltInQualityProfile {
+    private final Context context;
+    private final String name;
+    private final String language;
+    private boolean isDefault;
+    private final Map<RuleKey, NewBuiltInActiveRule> newActiveRules = new HashMap<>();
+
+    private NewBuiltInQualityProfileImpl(Context context, String name, String language) {
+      this.context = context;
+      this.name = name;
+      this.language = language;
+    }
+
+    @Override
+    public NewBuiltInQualityProfile setDefault(boolean value) {
+      this.isDefault = value;
+      return this;
+    }
+
+    @Override
+    public NewBuiltInActiveRule activateRule(String repoKey, String ruleKey) {
+      RuleKey ruleKeyObj = RuleKey.of(repoKey, ruleKey);
+      checkArgument(!newActiveRules.containsKey(ruleKeyObj), "The rule '%s' is already activated", ruleKeyObj);
+      NewBuiltInActiveRule newActiveRule = new NewBuiltInActiveRule(repoKey, ruleKey);
+      newActiveRules.put(ruleKeyObj, newActiveRule);
+      return newActiveRule;
+    }
+
+    @Override
+    public String language() {
+      return language;
+    }
+
+    @Override
+    public String name() {
+      return name;
+    }
+
+    @Override
+    public boolean isDefault() {
+      return isDefault;
+    }
+
+    @Override
+    public Collection<NewBuiltInActiveRule> activeRules() {
+      return newActiveRules.values();
+    }
+
+    @Override
+    public void done() {
+      checkArgument(isNotBlank(name), "Built-In Quality Profile can't have a blank name");
+      checkArgument(isNotBlank(language), "Built-In Quality Profile can't have a blank language");
+
+      context.registerProfile(this);
+    }
+
+    @Override
+    public String toString() {
+      StringBuilder sb = new StringBuilder("NewBuiltInQualityProfile{");
+      sb.append("name='").append(name).append('\'');
+      sb.append(", language='").append(language).append('\'');
+      sb.append(", default='").append(isDefault).append('\'');
+      sb.append('}');
+      return sb.toString();
+    }
+  }
+
+  interface BuiltInQualityProfile {
+    String name();
+
+    String language();
+
+    boolean isDefault();
+
+    @CheckForNull
+    BuiltInActiveRule rule(RuleKey ruleKey);
+
+    List<BuiltInActiveRule> rules();
+  }
+
+  @Immutable
+  class BuiltInQualityProfileImpl implements BuiltInQualityProfile {
+
+    private static final Logger LOG = Loggers.get(BuiltInQualityProfilesDefinition.BuiltInQualityProfileImpl.class);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "LOG" private field. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1068)